main.cpp

```diff
@@ -231,214 +333,320 @@ struct BasketballCourt
     std::string preferedProgrammingLanguage = "C++";
-    //    3) available funding (float)
     float availableFunding = 14121.5f;
-    //    4) prefered academic background (std::string)
     std::string preferedAcademicBackground = "Master's degree";
-    //    5) total number of programming languages used (int)
     int totalNumberOfProgrammingLanguagesUsed = 5;
-    //3 things it can do:
-    //    1) develop software
+
     void developSoftware();
-    //    2) implement new algorithms
     void implementNewAlgorithms();
-    //    3) reseach human perception of audio
     void researchHumanPerceptionOfAudio();
-
 };
 
-/*
-Thing 9) finance and advertising
+void RndDevelopment::developSoftware()
+
+{
+    bool researchFinished = true;
+    if(researchFinished == true)
+    {
+        std::cout << "start developing";
+    }
+    else
+    {
+        std::cout << "keep researching";
+    }    
+}
+
+void RndDevelopment::implementNewAlgorithms()
+{
+    bool algorithmEffective = true;
+    if(algorithmEffective == true)
+    {
+        std::cout << "implement into research";
+    }
+    else
+    {
+        std::cout << "keep optimizing";
+    }      
+}
+
+void RndDevelopment::researchHumanPerceptionOfAudio()
+{
+    int numberOfPsychoacousticTests = 12;
+    std::cout << "implement" << numberOfPsychoacousticTests << "tests";
+}
 
- */
 struct FinanceAndAdvertising
 {
-    //5 properties:
-    //    1) number of employees (int)
     int numberOfEmployees = 28;
-    //    2) employee salary (float)
     float employeeSalary = 62850.4f;
-    //    3) manager salary (float)
     float managerSalary = 81372.9f;
-    //    4) number of social media platforms used for advertising (int)
     int numberOfSocialMediaPlatformsUsedForAdvertising = 11;
-    //    5) department head (std::string)
     std::string departmentHead = "Mary";
-    //3 things it can do:
-    //    1) advertise the companie's products on social media
+
     void advertiseCompanyProductsOnSocialMedia();
-    //    2) create the company logo
     void createCompanyLogo();
-    //    3) pay the employees
-    float payTheEmployees(float employeeSalary, int numberOfEmployees); //returns the amount of money paid to the employes
-
+    float payTheEmployees(float employeeSalary, int numberOfEmployees);
 };
 
-/*
-Thing 10) audiovisual company
+void FinanceAndAdvertising::advertiseCompanyProductsOnSocialMedia()
+{
+    std::string mainPlatform = "platform";
+    int desiredViews = 19283;
+    std::cout << "advertise on" << mainPlatform << "until you reach" << desiredViews << "views";
+}
 
-*/
+void FinanceAndAdvertising::createCompanyLogo()
+{
+    std::string wordsOnLogo = "audiovisual";
+    std::cout << "create a logo based on" << wordsOnLogo;
+}
+
+float FinanceAndAdvertising::payTheEmployees(float employeesSalary, int totalNumberOfEmployees)
+{
+    float employeesFloat;
+    employeesFloat = float(totalNumberOfEmployees);
```

Comment:
Casting has not been covered, and this is a C-Style cast.
remove it
find another way to solve the warnings related to this type conversion.
- [ ] revise